### PR TITLE
[Pauli5] Add support in PauliWord and PauliSentence for qml.matrix 

### PR DESCRIPTION
**Pull request by Qottmann**
Not dependent on previous Pauli PRs ([Pauli1](https://github.com/PennyLaneAI/pennylane/pull/4989), [Pauli2](https://github.com/PennyLaneAI/pennylane/pull/5001), [Pauli3](https://github.com/PennyLaneAI/pennylane/pull/5003), [Pauli4](https://github.com/PennyLaneAI/pennylane/pull/5017)).

Add support in `qml.matrix` to handle `PauliWord` and `PauliSentence`

### CODE REVIEW

**Comment by Qottmann** [comment]
[sc-53282]

**Review by mudit2812 — APPROVED** [review]
Looks good. Thanks Korbinian!

File: pennylane/ops/functions/matrix.py [hunk 1]
```diff
@@ -176,6 +185,9 @@ def circuit():
 
     """
     if not isinstance(op, qml.operation.Operator):
+        if isinstance(op, (qml.pauli.PauliWord, qml.pauli.PauliSentence)):
```
[the remark this answers is not in the document]
Comment (Qottmann) [reply pennylane/ops/functions/matrix.py]: added ✅

[the remark this answers is not in the document]
Comment (Qottmann) [reply pennylane/ops/functions/matrix.py]: The additional branch raised a `too-many-branches` pylint error. I disabled that for the `matrix.py` file, that okay?

**Pull request merged**